### PR TITLE
[UIKit] [UIButton] Readd support for properly scaled resizeable images

### PR DESCRIPTION
**Pull request by davelamb**
This change removes the proxies for label and imageview, and uses actual labels and imageviews. It piggybacks on the UIScrollView work from Yi, but allowing you to specify a content, and layout element on a CALayerXaml. We grab a canvas element from the UIButton template, and use that for children layout; this allows us to addSubviews a UILabel and UIImageView to the Xaml button.

Tested with WOCCatalog, XAMLCatalog, and MM.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1320)

### CODE REVIEW

File: Frameworks/UIKit/UIButton.mm [hunk 1]
```diff
@@ -508,10 +480,6 @@ - (UIImage*)imageForState:(UIControlState)state {
 - (void)setTitleColor:(UIColor*)color forState:(UIControlState)state {
     _states[state].textColor = color;
     WUColor* convertedColor = ConvertUIColorToWUColor(color);
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIButton.mm]: Remove this line - no longer necessary

File: Frameworks/UIKit/UIButton.mm [hunk 2]
```diff
@@ -644,6 +612,15 @@ - (void)touchesCancelled:(NSSet*)touchSet withEvent:(UIEvent*)event {
...
+ @Status Interoperable
+*/
+- (void)setFrame:(CGRect)frame {
+    [super setFrame:frame];
+    [_xamlButton setWidth:self.bounds.size.width];
+    [_xamlButton setHeight:self.bounds.size.height];
+}
+
+/**
  @Status Stub
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIButton.mm]: Is this correctly annotated  - should it be caveat or stubbed with UNIMPLEMENTED()

File: Frameworks/UIKit/UIButton.mm [hunk 3]
```diff
@@ -850,19 +794,17 @@ - (UIColor*)currentTitleColor {
 }
 
 /**
- @Status Caveat
- @Notes Returns a mock UILabel that proxies some common properties and selectors to the underlying TextBlock
+ @Status Interoperable
 */
 - (UILabel*)titleLabel {
-    return (UILabel*)[[_proxyLabel retain] autorelease];
+    return [[_textLabel retain] autorelease];
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIButton.mm]: What is stopping us from making UIButton ARC-enabled like the other XAML-backed controls?

File: Frameworks/UIKit/UILabel.mm [hunk 4]
```diff
@@ -243,6 +243,9 @@ - (UIFont*)font {
  @Status Interoperable
 */
 - (void)setText:(NSString*)newStr {
+    if (newStr == nil && _text == nil) {
+        return;
+    }
     if (![newStr isKindOfClass:[NSString class]]) {
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UILabel.mm]: Newline above

File: Frameworks/UIKit/UILabel.mm [hunk 5]
```diff
@@ -243,6 +243,9 @@ - (UIFont*)font {
  @Status Interoperable
 */
 - (void)setText:(NSString*)newStr {
+    if (newStr == nil && _text == nil) {
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UILabel.mm]: Were you suppose to reference _text here in some way?

File: Frameworks/UIKit/UILabel.mm [hunk 6]
```diff
@@ -318,6 +321,9 @@ - (UITextAlignment)textAlignment {
  @Status Interoperable
 */
 - (void)setTextColor:(UIColor*)color {
+    if (color == nil && _textColor == nil) {
+        return;
+    }
     if (![_textColor isEqual:color]) {
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UILabel.mm]: Newline above

**Comment by davelamb** [comment]
Pull request deferred to https://github.com/Microsoft/WinObjC/pull/1361

**Pull request closed without merging**